engine/src/gfx/vdu.cpp

```diff
@@ -1185,21 +1185,18 @@ void VDU::DrawDamage(Unit *parent) {
     float x, y, w, h;
     //float th;
     //char st[1024];
-    GFXColor4f(1, parent->GetHull() / (*maxhull), parent->GetHull() / (*maxhull), 1);
+    double health_percent = parent->layers[0].facets[0].Percent();
+    GFXColor4f(1, health_percent, health_percent, 1);
     GFXEnable(TEXTURE0);
     float armor[8];
     parent->ArmorData(armor);
     const bool draw_damage_sprite = configuration()->graphics_config.hud.draw_damage_sprite;
     DrawHUDSprite(this, draw_damage_sprite ? parent->getHudImage() : NULL, .6, x, y, w, h,
```

Comment:
These lines are the bug. 
The code above was from a misreported line number, causing me to refactor GetHull.